mamba_ssm/ops/triton/layernorm.py

```diff
@@ -47,16 +47,25 @@ def rms_norm_ref(x, weight, bias, residual=None, eps=1e-6, prenorm=False, upcast
     out = out.to(dtype)
     return out if not prenorm else (out, x)
 
-
-@triton.autotune(
-    configs=[
+def config_prune(configs):
+    block_sz = 1024
+    # warp size is 64 in amd arch
+    warp_sz = 64 if torch.version.hip else 32
```

Comment:
warp size is not always 64 for amd.  could be 32 for radeon and 64 for instinct.  avoid hard-coding like this.  Though I don't seem to find a way to get the warp size from python torch like you can from the C++ torch side.  There is torch.cuda.get_device_properties(0), but only a limited set of them are mapped to python.  You can get gcnArchName from there and perhaps build a mapping from arch name to warp size.

Reply:
resolved